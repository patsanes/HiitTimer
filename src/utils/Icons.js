import React from 'react';
import { StyleSheet } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import IconI from 'react-native-vector-icons/Ionicons';
import IconA from 'react-native-vector-icons/AntDesign';
import IconE from 'react-native-vector-icons/Entypo';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  iconList: {
    alignSelf: 'center',
    marginRight: 10,
    marginLeft: 10,
    borderRadius: 3,
    borderWidth: 1,
    borderColor: 'white',
  },
});
export default class Icons extends React.Component {
  nested_Icons = name => {
    if (name == 'Countdown') {
      return <IconI name="ios-fitness" size={16} color="white" style={styles.iconList} />;
    }
    if (name == 'Rest') {
      return <IconA name="rest" size={16} color="white" style={styles.iconList} />;
    }
    if (name == 'Cycle') {
      return <IconE name="cycle" size={16} color="white" style={styles.iconList} />;
    }
    if (name == 'Serie') {
      return <Icon name="spinner" size={16} color="white" style={styles.iconList} />;
    }
    if (name == 'Rest between cycles') {
      return <IconI name="ios-timer" size={16} color="white" style={styles.iconList} />;
    }
    return <IconE name="stopwatch" size={16} color="white" style={styles.iconList} />;
  };
  render() {
    const { name } = this.props;
    return this.nested_Icons(name);
  }
}

Icons.propTypes = {
  name: PropTypes.string.isRequired,
};
