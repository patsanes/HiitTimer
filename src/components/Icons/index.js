import React from 'react';
import { StyleSheet } from 'react-native';
import IconFontAwesome from 'react-native-vector-icons/FontAwesome';
import IconIonicons from 'react-native-vector-icons/Ionicons';
import IconAntDesign from 'react-native-vector-icons/AntDesign';
import IconEntypo from 'react-native-vector-icons/Entypo';
import PropTypes from 'prop-types';
import { colors, fontSizes, space } from 'HiitTimer/src/utils/theme';


const styles = StyleSheet.create({
  iconList: {
    width: space.xxLarge,
    textAlign: 'center',
  },
});

const iconsSettings = {
  Countdown: { name: 'ios-fitness', Icon: IconIonicons },
  Rest: { name: 'rest', Icon: IconAntDesign },
  Cycle: { name: 'cycle', Icon: IconEntypo },
  Serie: { name: 'spinner', Icon: IconFontAwesome },
  'Rest between cycles': { name: 'ios-timer', Icon: IconIonicons },
  'Start countdown': { name: 'stopwatch', Icon: IconEntypo },
  'chevron-up': { name: 'chevron-up', Icon: IconFontAwesome },
};

const Icons = ({ name: iconName, size: sizeIcon, color: colorIcon }) => {
  const { name, Icon } = iconsSettings[iconName];
  return <Icon name={name} size={sizeIcon} color={colorIcon} style={styles.iconList} />;
};

Icons.propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.number,
  color: PropTypes.string,
};
Icons.defaultProps = {
  size: fontSizes.xxSmall,
  color: colors.secondaryDark,
};

export default Icons;
