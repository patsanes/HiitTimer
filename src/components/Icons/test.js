import React from 'react';
import { render } from 'react-native-testing-library';

import Icons from './index';

describe('Icons', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<Icons name="chevron-up" size={30} color="white" />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly using default color', () => {
    const { toJSON } = render(<Icons name="chevron-up" size={30} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly using default size', () => {
    const { toJSON } = render(<Icons name="chevron-up" color="white" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
