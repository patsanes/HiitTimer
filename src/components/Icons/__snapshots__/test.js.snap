// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons renders correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "white",
        "fontSize": 30,
      },
      Object {
        "textAlign": "center",
        "width": 30,
      },
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Icons renders correctly using default color 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#373B44",
        "fontSize": 30,
      },
      Object {
        "textAlign": "center",
        "width": 30,
      },
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Icons renders correctly using default size 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "white",
        "fontSize": 16,
      },
      Object {
        "textAlign": "center",
        "width": 30,
      },
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;
